daemon/images/image_list.go

```diff
@@ -53,8 +53,8 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 		}
 		// Resolve multiple values to the oldest image,
 		// equivalent to ANDing all the values together.
-		if beforeFilter.IsZero() || beforeFilter.After(img.Created) {
-			beforeFilter = img.Created
+		if img.Created != nil && (beforeFilter.IsZero() || beforeFilter.After(*img.Created)) {
+			beforeFilter = *img.Created
```

Comment:
In the interest of highlighting the bits I'm the least sure about, this is one I _think_ I've got right, but am not positive.

Reply:
@vvoland @laurazard could one of you look if there's any possible concerns? (this comment, and the one below)

Reply:
One thing that concerns me here is the behavior disparity between `Created = nil` and `Created = time.Time{}`.
Indeed, with the current model (where `Created` is a pointer), it technically makes sense for `nil` and zero value have different meanings.
However, in practice, I _think_ (but please correct me if I'm making things up) it's actually more likely for  `Created` to be set to `1970-01-01 ...` when the created time is not known/explicitly discarded (for example images build by buildkit with `SOURCE_DATE_EPOCH=0`).

So, personally I think this PR behavior for `img.Created = nil` is correct (ignore filter) and we should change the `Created.IsZero()` case to do the same. It doesn't make sense to filter images created before Unix epoch, because no image could be created then, so it most real-world cases it has the same meaning as `nil`.

Reply:
I think this is correct. Both `Created = nil` and `Created = 0` will cause the filter to be effectively ignored and won't filter out any image.

Although with the current model (where `Created` is a pointer), it technically makes sense for `nil` and zero value have different meanings, in practice, I _think_ (but please correct me if I'm making things up) it's actually more likely for  `Created` to be set to `1970-01-01 ...` when the created time is not known/explicitly discarded (for example images build by buildkit with `SOURCE_DATE_EPOCH=0`).

It doesn't make sense to filter images created before Unix epoch, because no image could be created then, so it most real-world cases it would have the same meaning as `nil`.

Reply:
ISTR there also was some magic "reproducible build" date, but can't find it right now (I do recall we had https://github.com/docker/cli/pull/2107, which uses `.IsZero()`)

(looked at https://reproducible-builds.org/specs/source-date-epoch/ and https://reproducible-builds.org/docs/source-date-epoch/, but can't find it at a glance)

Reply:
Yeah, I honestly took a sec to review this because the first time I looked at these lines I was like "Oh, I should take a closer look" and then totally forgot 😅  I (think) it looks accurate though :)